docs/fastdds/dds_layer/publisher/dataWriterListener/dataWriterListener.rst

```diff
@@ -3,3 +3,40 @@
 DataWriterListener
 ==================
 
+:class:`DataWriterListener` is an abstract class defining the callbacks that will be triggered
+in response to state changes on the :ref:`dds_layer_publisher_dataWriter`.
+By default, all these callbacks are empty and do nothing.
+The user should implement a specialization of this class overriding the callbacks
+that are needed on the application.
+Callbacks that are not overridden will maintain their empty implementation.
+
+:class:`DataWriterListener` defines the following callbacks:
+
+ * **on_publication_matched**: The :ref:`dds_layer_publisher_dataWriter` has found a
+   :ref:`dds_layer_subscriber_dataReader` that matches the :ref:`dds_layer_topic_topic` and has
+   a common partition and a compatible QoS, or has ceased to be matched with a
+   :ref:`dds_layer_subscriber_dataReader` that was previously considered to be matched.
+
+ * **on_offered_deadline_missed**: The :ref:`dds_layer_publisher_dataWriter` failed to provide
+   data within the deadline period configured on its :ref:`dds_layer_publisher_dataWriterQos`.
+   It will be called for each deadline period and data instance for which the
+   :ref:`dds_layer_publisher_dataWriter` failed to provide data.
+
+ * **on_offered_incompatible_qos**: The :ref:`dds_layer_publisher_dataWriter` has found a
+   :ref:`dds_layer_subscriber_dataReader` that matches the :ref:`dds_layer_topic_topic` and has
+   a common partition, but with a requested QoS that is incompatible with the one defined on the
+   :ref:`dds_layer_publisher_dataWriter`.
+
+   Currently this callback is not implemented (it will never be called), and will be implemented
```

Comment:
In the HTML it's not clean whether the one not implemented is `on_offered_incompatible_qos` or `on_liveliness_lost`